engine/classes/ElggFile.php

```diff
@@ -80,6 +80,18 @@ public function getFilenameOnFilestore() {
 	}
 
 	/**
+	 * Returns path relative to the filestore (data directory) root
+	 *
```

Comment:
Let's solicit some feedback from the community. I am not sure how usable that API is in practice. Setting a custom filestore for individual files doesn't seem like something that would work in practice, unless you hack `ege*` with custom callback function that would inject custom filestore.